[RS] - Filter Array Collections

rysmith opened the pull request:
### Summary
Currently Brita only supports filtering of Active Record collections.  While this works for most use cases, there are times when it is necessary to filter plain ruby arrays.  This PR offers basic support for filtering arrays.  

#### Supported Types
* int - Sort on an integer column
* decimal - Sort on a decimal column
* string - Sort on a string column
* text - Sort on a text column

#### Sorting
Sorting is not supported at this time.

#### Testing
- [ ] Filtering and Sorting of AR collections should continue to function as normal.
- [ ] Basic filtering should be supported for the above listed types.
- [ ] Int types should be able to filter with multiple values `filter[id]=[1,2]`

HParker commented on lib/brita/collection/wrapper.rb:
```diff
@@ -0,0 +1,19 @@
+module Brita
+  module Collection
+    class Wrapper
```
Should this be `ArrayWrapper`?

HParker commented on test/dummy/app/controllers/posts_controller.rb:
```diff
@@ -41,6 +41,12 @@ def index
+  def list
+    posts = Post.all.map(&:attributes)
+
+    render json: filtrate(posts)
```
Ah, that is a neat way to do this test.

rysmith replied:
Thanks! :)

rysmith commented:
@HParker I've made the update to readme so it's clear we're talking about the array that is passed into `filtrate(..)` and the next code block is the example. 

I've left the module name as `Collection` and changed the class names to reflect the fact that the expect to deal with Arrays: `ArrayWrapper`, `ArrayFilter` and `ArraySort`.  This will allow for other types of objects to be wrapped but all under the generic `Collection` module. 

I've also updated the naming convention to `...Filter` and `...Sort` which then implement `where(..)` and `order(..)`.

Let me know what you think.

amayer171 commented on test/controller_test.rb:
```diff
@@ -282,4 +283,48 @@ class PostsControllerTest < ActionDispatch::IntegrationTest
+    json = JSON.parse(@response.body)
+
+    assert_equal [filtered_post.id], (json.map { |post| post["id"] })
+  end
```
You may want to also add a test for filtering on ranges. I think you could do it with a decimal example:
`filters: { rating: 5.0...6.3 }`

Looks like your changes don't support ranges though? In either case it may be nice to have a spec that shows that it does/ doesn't support ranges.

rysmith replied:
Agreed, I'll see how hard it will be to implement, but I'll get a test in either way.

rysmith commented:
Thanks for the feedback on this.  Unfortunately we've decided to go a different direction and don't need the non-AR collection sorting anymore.
